mavros_extras/src/plugins/mount_control.cpp

```diff
@@ -209,13 +242,24 @@ class MountControlPlugin : public plugin::PluginBase {
 	 */
 	void handle_mount_orientation(const mavlink::mavlink_message_t *msg, mavlink::common::msg::MOUNT_ORIENTATION &mo)
 	{
-		auto q = ftf::quaternion_from_rpy(Eigen::Vector3d(mo.roll, mo.pitch, mo.yaw) * M_PI / 180.0);
+		const auto timestamp = ros::Time::now();
+		// some gimbals send negated/inverted angle measurements, correct that to obey the MAVLink frame convention
+		if (negate_measured_roll) {
+			mo.roll = -mo.roll;
+		}
+		if (negate_measured_pitch) {
+			mo.pitch = -mo.pitch;
+		}
+		if (negate_measured_yaw) {
+			mo.yaw = -mo.yaw;
+			mo.yaw_absolute = -mo.yaw_absolute;
+		}
```

Comment:
This will make the frames inconsistent (oscillating between left hand coordinate or right hand coordinate systems). Shouldn't this be something that is fixed on the driver level?

Reply:
Gremsy probably behaves differently from storm32. I can not test I only have greasy. But what I can say ist that this let's the users fix issues on their Gimbals without changing the code.

Reply:
If this is a device specific fix that needs to happen, why not handle it on the user app level? I still think mavros is the wrong place to make a workaround for a device specific problem.

Reply:
Because the diagnostics code needs to handle it at this level, take a look at the diagnostics class in the mount_control.cpp file

Reply:
Not sure if I understand what you mean... why does the diagnostic status influenced by the negation in the mount control handling?

Reply:
@amilcarlucas But if the gremsy gimbal driver is implemented the opposite way, shouldn't the user app command -70° in the first place?

Reply:
**That** is the gremsy bug, you command 70 and it measures -70 !